extensions/binding/org.eclipse.smarthome.binding.mqttbroker/src/main/java/org/eclipse/smarthome/binding/mqttbroker/internal/MqttWillAndTestamentBuilder.java

```diff
@@ -0,0 +1,65 @@
+/**
+ * Copyright (c) 2010-2017 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.smarthome.binding.mqttbroker.internal;
+
+import org.apache.commons.lang.StringUtils;
+import org.eclipse.smarthome.io.transport.mqtt.MqttWillAndTestament;
+
+/**
+ * Builds a {@link MqttWillAndTestament}.
+ *
+ * @author David Graeff - Initial contribution
+ */
+public class MqttWillAndTestamentBuilder {
+    private String topic;
+    private byte[] message;
+    private int qos;
+    private boolean retain;
+
+    public static MqttWillAndTestamentBuilder create(String topic) {
+        if (StringUtils.isBlank(topic)) {
+            throw new IllegalArgumentException("Topic must not be null");
```

Comment:
Null could be prevented by an annotation, but you seem to also require it to be non-empty, so the message should state so. Btw, JavaDoc would be nice on that method!